lib/memery.rb

```diff
@@ -35,10 +35,12 @@ def included(base = nil, &block)
   extend ModuleMethods
 
   module ClassMethods
-    def memoize(method_name, condition: nil, ttl: nil)
+    def memoize(*method_names, condition: nil, ttl: nil)
       prepend_memery_module!
-      define_memoized_method!(method_name, condition: condition, ttl: ttl)
-      method_name
+      method_names.each do |method_name|
+        define_memoized_method!(method_name, condition: condition, ttl: ttl)
+      end
+      method_names.length > 1 ? method_names : method_names.first
```

Comment:
Method-chaining was the only awkwardness - I assume we want to preserve the `foo bar memoize def baz` decorator style, but when memoizing multiple methods we probably want to return the lot. I generally prefer type-signatures not to be dynamic in this way, but it seemed like the least-problematic result.

Reply:
This looks OK for me.